File: 프린터.java
import java.util.*;
class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Deque<ArrayList<Integer>> d = new ArrayDeque<>();
        int cnt = 0;
        for(int i : priorities){
            ArrayList<Integer> arr = new ArrayList<>();
            arr.add(cnt);
            arr.add(i);
            d.add(arr);
            cnt++;
        }
        while(d.size()>0){
            ArrayList<Integer> now = d.pollFirst();
            int maxi = -1;
            Iterator<ArrayList<Integer>> i = d.iterator();
            while(i.hasNext()){
                int temp= i.next().get(1);
                if(temp > maxi) maxi = temp;    
            }
            if(now.get(1) < maxi){
                d.add(now);
            }else{
                answer++;
                if(location == now.get(0)) break;
            }
            
        }
        return answer;
    }
}

File: 프린터.py
from collections import deque
def solution(priorities, location):
    answer = 0
    arr = deque()
    for i in range(len(priorities)):
        arr.append((priorities[i], i))
    while arr:
        top = arr.popleft()
        if arr and top[0] < max(arr)[0]:
            arr.append(top)
        else:
            answer+=1
            if top[1] == location:
                break
    return answer
